### PR TITLE
feat: new cosmos keyring helper for injective client

### PR DESCRIPTION
**Pull request by maxim-inj**
Hi, a little backstory here. Original Cosmos keyring helper was created for [Peggo CLI](https://github.com/InjectiveLabs/peggo/commit/dd8648d658ce4418eeb1a499dcc7407199ea6114#diff-1cdc0d8796f6c5f17e6ec0c51db480c6b4a72869f1e691729ed89481762af448L41) service. It was very cool and really useful, so once we wanted to use the same approach in other apps as well ([oracles](https://github.com/InjectiveLabs/injective-price-oracle/blob/9cf9e00f98833c1326938d26b82671b445920863/cmd/injective-price-oracle/oracle.go#L132-L139
), etc), we also copied the piece into [sdk-go](https://github.com/InjectiveLabs/sdk-go/blob/b7a68cc0be42f5bdd9476ebc4717efa0053698c9/client/chain/keys.go#L28-L36) and it sat there for a couple of years.

The fact is that it was never properly tested (unit tests for all branches), it never had interface of possible errors for all failing conditions and it also had a couple of other bugs. In general, the problem with that keyring design was that if the app wants to have 2+ keys loaded at the start time, you can't add them the same way, only using multiple keyrings. Which is not cool.

Last 2 years I was actually reusing the same keyring drop-in function, but during the last year decided to sit down and refactor it, cover with tests and changed the design a bit.

So here is the new version of that keyring helper. I think this PR can be merged as-is once approved, but it will take some time to migrate dependant apps to the new keyring implementation. The latter is not urgent nor necessary. Given that peggo config is very mission critical, I'd advise not change anything in its refactored code.

Feel free to check it out and see the [client/keyring/README.md](https://github.com/InjectiveLabs/sdk-go/blob/d60376e731d96fa846bbe8f3cd9aa31289f323e8/client/keyring/README.md) for more details.

## Summary by CodeRabbit

- **New Features**
	- Introduced comprehensive documentation for the "Injective Chain Keyring Helper," detailing keyring creation and management for the Cosmos SDK.
	- Implemented a keyring management system that supports various key types and configurations, enhancing secure blockchain interactions.
	- Added validation for key configurations, ensuring compatibility and security during keyring operations.

- **Bug Fixes**
	- Enhanced error handling for keyring operations, providing clearer messages for various failure scenarios.

- **Tests**
	- Added a suite of unit tests for the keyring functionality, validating error handling and key management processes.

- **Chores**
	- Updated project dependencies to include new libraries for improved functionality.
	- Refined pre-commit configuration to exclude test data directories from checks.
	- Enhanced GitHub Actions workflow for secure handling of sensitive data during testing.

### CODE REVIEW

File: client/keyring/errors.go
Comment (maxim-inj) [reply client/keyring/errors.go:20]: >  For example, all error names could start with ErrKeyring to clearly associate them with the keyring context.

This rabbit doesn't know that in [Go naming practices](https://go.dev/blog/package-names) the package should provide such context, as for external user the error would be like `keyring.ErrUnexpectedAddress`

File: client/keyring/keyring_config.go
Comment (maxim-inj) [reply client/keyring/keyring_config.go:63]: > redundant check for a non-empty Backend value.
> is a string type and its zero value (empty string) is a valid state

no way, rabbit

**Comment by aarmoa** [comment]
Hello @maxim-inj. I will start reviewing the PR. In the meantime, could you please do the following?
- Point the PR to the `dev` branch. Unless we are working on a urgent fix, all PRs should be created pointing to `dev`
- Make sure that both the `pre-commit` and `run-tests` workflows pass in green

**Review by aarmoa — COMMENTED** [review]
This is great @maxim-inj, thank you for adding this new implementation to the SDK. 
I think it is great that the new keyring is in  a new module, so that we can support the old and new implementations in parallel. Once the PR is merged we should create a task to migrate all repo examples to use the new keyring, and also to mark the old keyring creation logic as deprecated.

I will be happy to approve the PR once all the validation workflows pass in green

**Comment by maxim-inj** [comment]
> Unless we are working on a urgent fix, all PRs should be created pointing to dev

Oh didn't know, indeed. Rebased, will wait for CI to pass.

> Once the PR is merged we should create a task to migrate all repo examples to use the new keyring, and also to mark the old keyring creation logic as deprecated.

Yeah this will be an ongoning process with low priority.

**Comment by aarmoa** [comment]
@maxim-inj still waiting for you to resolve all issues in the validation workflows before approving the PR

**Comment by maxim-inj** [comment]
@aarmoa alright it's good to go!

**Review by aarmoa — APPROVED** [review]
Changes look good to me

**Pull request merged**
